File: 2021/src/test/java/de/jfriemel/aoc/days/Day14Test.java
package de.jfriemel.aoc.days;

import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class Day14Test {

    private static final String INPUT_STRING = """
            NNCB
                        
            CH -> B
            HH -> N
            CB -> H
            NH -> C
            HB -> C
            HC -> B
            HN -> C
            NN -> C
            BH -> H
            NC -> B
            NB -> B
            BN -> B
            BB -> N
            BC -> B
            CC -> N
            CN -> C""";

    private static final List<String> INPUT = Arrays.asList(INPUT_STRING.split("\n"));

    @Test
    public void testPart1() {
        String result = new Day14().part1(INPUT);
        assertEquals("1588", result);
    }

    @Test
    public void testPart2() {
        String result = new Day14().part2(INPUT);
        assertEquals("2188189693529", result);
    }

}

File: 2023/src/test/kotlin/de/jfriemel/aoc/days/Day14Test.kt
package de.jfriemel.aoc.days

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class Day14Test {
    private val input = """
        O....#....
        O.OO#....#
        .....##...
        OO.#O....O
        .O.....O#.
        O.#..O.#.#
        ..O..#O..O
        .......O..
        #....###..
        #OO..#....
    """.trimIndent().lines()

    @Test
    fun testPart1() {
        val result = Day14.part1(input)
        Assertions.assertEquals("136", result)
    }

    @Test
    fun testPart2() {
        val result = Day14.part2(input)
        Assertions.assertEquals("64", result)
    }
}
